webui/src/lib/onboarding/repoOnboardingService.tsx

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable react/jsx-no-target-blank */
+import React from "react";
+import { OnboardingStep } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects, NotFoundError } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_OBJECT_PATH = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  try {
+    await objects.head(repo, defaultBranch, SAMPLE_REPO_OBJECT_PATH);
+    return true;
+  } catch (e) {
+    return false;
+  }
+  
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const promises = [
+    retention.setGCPolicyPreflight(repoId),
+    branchProtectionRules.createRulePreflight(repoId),
+    objects.uploadPreflight(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    ),
+  ];
+  
+  try {
+    await Promise.all(promises);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+export const getRepoOnboardingSteps = (
+  currentRepo: string,
+  defaultBranch: string,
+  objectStoreName: string,
+  navigate: NavigateFunction
+): OnboardingStep[] => [
+  {
+    id: "import-data",
+    title: "Import your existing data",
+    description: (
+      <>
+        <p>Import your existing data from ${objectStoreName} to your repository by creating pointers to it (no copying of data involved).</p>
+        <p>Please see our documentation for information on the <a href="https://docs.lakefs.io/howto/import.html#limitations" target="_blank">limitations of the import process</a> and <a href="https://docs.lakefs.io/howto/import.html#working-with-imported-data" target="_blank">working with imported data</a></p>,
+      </>
+    ),
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo, "_");
```

Comment:
```suggestion
      const repoBranches = await branches.list(currentRepo, "_", 1000);
```
perhaps after making "1000" a constant LAKEFS_MAX_PAGE_SIZE or something.

Listing by prefix can return up to 1000 elements per page.  The webui API client limits itself to **100** objects per page, because it _usually_ makes little sense to return huge pages to a GUI.  But _here_ there is little point in asked for _fewer_ branches, so let's ask for the most he server can return!

Reply:
Done